File: Data Structures & Algorithms/kth-largest-integer-in-a-stream/submission-1.cpp
class KthLargest {
public:
    priority_queue<int>pq;
    int kk;
    KthLargest(int k, vector<int>& nums) {
        kk  = k;
        for(int i=0;i<nums.size();i++){
            if(pq.size()<k){
                pq.push(-(nums[i]));
            }
            else{
                if(-pq.top()<nums[i]){
                    pq.pop();
                    pq.push(-nums[i]);
                }
            }

        }

    }
    
    int add(int val) {
        if(pq.size()<kk){
            pq.push(-val);
        }
        else{
            if(-pq.top()<val){
                    pq.pop();
                    pq.push(-val);
            }
        }
        return -pq.top();
    }
};

File: Data Structures & Algorithms/subsets/submission-1.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:

        def req(nums,i,curr):
            nonlocal ans
            if len(nums) == i:
                ans.append(curr.copy())
                return
            req(nums,i+1,curr)
            curr.append(nums[i])
            req(nums,i+1,curr)
            curr.pop()
            return

        ans = []
        curr = []
        req(nums,0,curr)
        return ans

        